src/transport/transport.h

```diff
@@ -92,15 +92,22 @@ class ConvectionTransport : public TransportBase {
         FieldSet output_fields;
     };
 
+
+    typedef ConcentrationTransportBase FactoryBaseType;
+
+    static const Input::Type::Record & get_input_type();
+
     /**
      * Constructor.
      */
-        ConvectionTransport(Mesh &init_mesh, const Input::Record in_rec);
+    ConvectionTransport(Mesh &init_mesh, const Input::Record in_rec);
 	/**
 	 * TODO: destructor
 	 */
 	virtual ~ConvectionTransport();
 
+	void initialize() override;
```

Comment:
The instance of ConcentrationTransportBase (CTB) is constructed in 3 steps:
1) TransportOperatorSplitting (TOS) calls the constructor
2) TOS initializes several members of CTB from its input record
3) CTB::initialize() finalizes the construction.
If the constructor takes an additional argument, the input record of TOS, then it may be possible to eliminate initialize().